tests/expectations/parser-statement/statement/conditional_fail.out

```diff
@@ -1,5 +1,5 @@
-Error [EPAR0370005]: expected : -- found '='
-    --> test_0:1:23
+Error [EPAR0370017]: Could not parse the implicit value: 2.
+    --> test_0:1:25
      |
    1 | if true {} else let x = 2;
```

Comment:
Should inference also take care of numeric literals?

Reply:
Yes. I already opened this issue https://github.com/ProvableHQ/leo/issues/28626 which I'm working on next. All this PR is doing is type inference for variables.